serialization/untyped_boolean.go

```diff
@@ -0,0 +1,23 @@
+package serialization
+
+import "errors"
+
+// UntypedBoolean defines an untyped boolean object.
+type UntypedBoolean struct {
+	UntypedNode
+}
+
+// GetValue returns the bool value.
+func (un *UntypedBoolean) GetValue() *bool {
+	return un.value.(*bool)
+}
+
+// NewUntypedBoolean creates a new UntypedBoolean object.
+func NewUntypedBoolean(boolValue *bool) (*UntypedBoolean, error) {
```

Comment:
we're probably going to get feedback along the line of this being really convoluted in GO:
1. accept a reference to a bool
2. if the reference is nil, return an error

We should probably accept the boolean value directly, unless there's another constraint I'm not seeing?
And then we can make a reference to that value in the body to facilitate the later type assertion in the getter.
This way the constructor doesn't need to return an error anymore.

Reply:
Thanks for the pointer here. Really makes things clearer.